source/module_base/complexarray.h

```diff
@@ -83,23 +89,37 @@ class ComplexArray
 ComplexArray operator*(double r, const ComplexArray &cd);
 ComplexArray operator*(std::complex < double> c, const ComplexArray &cd);
 
+//Sum of absolute squares of all elements in cd
 double abs2(const ComplexArray &cd);
 
+
 void
 add_scale_abs2(const std::complex < double> &c, const ComplexArray & in,
                ComplexArray &out);
 
+// Take "dot-product" of two ComplexArray:  sum of cd1(conjugate)[i] * cd2[i]
 std::complex < double> dot(const ComplexArray &cd1, const ComplexArray &cd2);
+
+/* Does cd2 += r * cd1 */
 void scale_accumulate(double r, const ComplexArray &cd1, ComplexArray &cd2);
+
+/* Does cd2 += c * cd1 */
 void scale_accumulate(std::complex < double> c, const ComplexArray &cd1, ComplexArray &cd2);
```

Comment:
If possible, add `const` for those parameters not modified.

Reply:
it will modify cd2

Reply:
> it will modify cd2

Actually `c` I mean